src/Type/FloatType.php

```diff
@@ -149,6 +150,33 @@ public function traverse(callable $cb): Type
 		return $this;
 	}
 
+	public function exponentiate(Type $exponent): Type
+	{
+		$numberType = new UnionType([
+			new IntegerType(),
+			new FloatType(),
+			TypeCombinator::intersect(
+				new StringType(),
+				new AccessoryNumericStringType(),
+			),
+		]);
+
+		if ($exponent instanceof NeverType) {
+			return new NeverType();
+		}
+		if (!$numberType->isSuperTypeOf($exponent)->yes()) {
+			return new ErrorType();
+		}
+
+		if ($exponent instanceof ConstantScalarType) {
+			if ($exponent->getValue() == 0) {
```

Comment:
the cs-fixer will turn these into strict comparisons, which in turn breaks a lot of tests.
I think we should ignore these from the cs fixer